src/disk_interface.h

```diff
@@ -44,9 +44,18 @@ struct FileReader {
 /// Abstract so it can be mocked out for tests.  The real implementation
 /// is RealDiskInterface.
 struct DiskInterface: public FileReader {
+  /// How to treat symlinks when calling Stat().
+  enum SymlinkTreatment {
+    kFollow,      // Used when the only the file contents relevant.
+    kDontFollow,  // Used for outputs.
+    kTakeNewest,  // Used for inputs.
```

Comment:
Thanks for the comments here, they're very helpful.

However, aren't most nodes both outputs and inputs, yet stat()'d only once? What happens in that case?